### PR TITLE
Update the APIRule config

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Update the APIRule config.

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/17371.

### CODE REVIEW

File: components/eventing-controller/controllers/backend/reconciler.go [hunk 1]
```diff
@@ -348,10 +350,24 @@ func (r *Reconciler) syncOauth2ClientIDAndSecret(ctx context.Context, backendSta
 		r.credentials.clientID = credentials.clientID
 		r.credentials.clientSecret = credentials.clientSecret
 		r.credentials.tokenURL = credentials.tokenURL
+		certsURL := computeCertsURL(string(credentials.tokenURL))
+		r.credentials.certsURL = []byte(certsURL)
 	}
 	return nil
 }
 
+// computeCertsURL computes the certs url from the given token url.
```
Comment (muralov) [inline components/eventing-controller/controllers/backend/reconciler.go:359]: Is it not good idea to add certs URL to the `eventing-webhook-auth` secret first? Then our code didn't this addition logic and no additional tests. It leads in general less work. Hence, I'd suggest to pause this.

Comment (marcobebway) [reply components/eventing-controller/controllers/backend/reconciler.go]: Changed to read the certs url directly from the secret.

File: components/eventing-controller/pkg/subscriptionmanager/eventmesh/eventmesh.go [hunk 2]
```diff
@@ -248,23 +248,32 @@ func getOAuth2ClientCredentials(params subscriptionmanager.Params) (*backendeven
...
 	}
+
 	val = params[subscriptionmanager.ParmaNameTokenURL]
 	tokenURL, ok := val.([]byte)
 	if !ok {
-		return nil, fmt.Errorf("expected []byte value for %s, but received %T", subscriptionmanager.ParmaNameTokenURL, val)
+		return nil, fmt.Errorf("expected []byte value for %s", subscriptionmanager.ParmaNameTokenURL)
+	}
+
+	val = params[subscriptionmanager.ParmaNameCertsURL]
```
Comment (muralov) [inline components/eventing-controller/pkg/subscriptionmanager/eventmesh/eventmesh.go:266]:
```suggestion
	val = params[subscriptionmanager.ParamNameCertsURL]
```

Comment (muralov) [reply components/eventing-controller/pkg/subscriptionmanager/eventmesh/eventmesh.go:266]: (nitpick) correcting spelling error

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go [hunk 3]
```diff
@@ -175,6 +176,288 @@ func TestReconciler_Reconcile(t *testing.T) {
...
+
+			// then
+			require.Equal(t, tc.wantHandler.Name, apiRule.Spec.Rules[0].AccessStrategies[0].Handler.Name)
+			require.Equal(t, tc.wantHandler.Config, apiRule.Spec.Rules[0].AccessStrategies[0].Handler.Config)
+		})
+	}
+}
+
+// TestReconciler_APIRuleConfig_Upgrade ensures that the created APIRule is configured correctly
+// before and after the upgrade from ory to Eventing webhook auth and vise versa.
```
Comment (muralov) [inline components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go:286]: I thought we will not downgrade to ory. Is it a valid use case?

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go [hunk 4]
```diff
@@ -175,6 +176,288 @@ func TestReconciler_Reconcile(t *testing.T) {
...
+			},
+			wantHandlerAfterUpgrade: apigatewayv1beta1.Handler{
+				Name: object.OAuthHandlerNameJWT,
+				Config: &runtime.RawExtension{
+					Raw: []byte(fmt.Sprintf(object.JWKSURLFormat, credentials.CertsURL)),
+				},
+			},
+		},
+		{
+			name:              "Eventing webhook auth is enabled before the upgrade",
```
Comment (muralov) [inline components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go:341]: please see the [comment](https://github.com/kyma-project/kyma/pull/17632/files#r1229571234)

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go (hunk 3 above)
Comment (marcobebway) [reply components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go:286]: As long as we have a feature flag, we must test switching it on and off. In this case, it means deciding to use which secret.

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go (hunk 4 above)
Comment (marcobebway) [reply components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go:341]: Please check https://github.com/kyma-project/kyma/pull/17632#discussion_r1229603629

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go (hunk 3 above)
Comment (muralov) [reply components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go:286]: as far as I understand, the goal of the feature flag is to not run your new code until other side is ready. If other side were ready, you'd we'd always enable feature flag as ORY will be deleted soon. Hence, I don't see a real use case for going back. 
Although having the test case is little bit more code, but  it is minor thing. Hence, you can keep the test case too.

**Review by muralov — DISMISSED** [review]

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go (hunk 3 above)
Comment (marcobebway) [reply components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go:286]: In genera l agreed with your comment, but also the edge cases should be tested, and IMO switching the feature flag on/off (back and forth) should be tested.

**Review by muralov — APPROVED** [review]

**Pull request merged**